java/java.file.launcher/src/org/netbeans/modules/java/file/launcher/SingleSourceFileUtil.java

```diff
@@ -116,6 +118,10 @@ public static Process compileJavaSource(FileObject fileObject, JavaPlatform jdk)
         if (!vmOptions.isEmpty()) {
             compileCommandList.addAll(Arrays.asList(vmOptions.split(" "))); //NOI18N
         }
+        vmOptions = NbPreferences.forModule(SingleSourceFileUtil.class).get("vmOptions", "").trim(); // NOI18N
```

Comment:
this could be extracted since it is a constant which is used in 3 files already.

This file has similar constants at L68, please rename to `globalVmOptions` or possibly even use the same naming pattern as the other keys but prefix with `global_`.